android/app/src/main/java/org/fossasia/openevent/core/discount/DiscountCodeFragment.java

```diff
@@ -0,0 +1,155 @@
+package org.fossasia.openevent.core.discount;
+
+import android.arch.lifecycle.ViewModelProviders;
+import android.content.Intent;
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.support.design.widget.Snackbar;
+import android.support.v4.widget.SwipeRefreshLayout;
+import android.support.v7.widget.DefaultItemAnimator;
+import android.support.v7.widget.LinearLayoutManager;
+import android.support.v7.widget.RecyclerView;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import org.fossasia.openevent.R;
+import org.fossasia.openevent.common.network.NetworkUtils;
+import org.fossasia.openevent.common.ui.base.BaseFragment;
+import org.fossasia.openevent.common.utils.Utils;
+import org.fossasia.openevent.core.auth.AuthUtil;
+import org.fossasia.openevent.core.auth.LoginActivity;
+import org.fossasia.openevent.data.DiscountCode;
+
+import java.lang.ref.WeakReference;
+import java.util.ArrayList;
+import java.util.List;
+
+import butterknife.BindView;
+import timber.log.Timber;
+
+public class DiscountCodeFragment extends BaseFragment {
+
+    private List<DiscountCode> discountCodes = new ArrayList<>();
+    private DiscountCodesListAdapter discountCodesListAdapter;
+
+    @BindView(R.id.discount_refresh_layout)
+    SwipeRefreshLayout swipeRefreshLayout;
+    @BindView(R.id.txt_no_discount_codes)
+    TextView noDiscountCodeView;
+    @BindView(R.id.discount_code_header)
+    protected TextView discountCodeHeader;
+    @BindView(R.id.list_discount_codes)
+    protected RecyclerView discountCodesRecyclerView;
+    private DiscountFragmentViewModel discountFragmentViewModel;
+
+
+    @Nullable
+    @Override
+    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
+        setHasOptionsMenu(true);
+
+        final View view = super.onCreateView(inflater, container, savedInstanceState);
+        setUpRecyclerView();
+        Utils.registerIfUrlValid(swipeRefreshLayout, this, this::refresh);
+        //set up view model
+        discountFragmentViewModel = ViewModelProviders.of(this).get(DiscountFragmentViewModel.class);
+        if(AuthUtil.isUserLoggedIn()){
+            if(NetworkUtils.haveNetworkConnection(getContext())) {
+                swipeRefreshLayout.setRefreshing(true);
+                downloadDiscountCodes();
+            }
+            loadData();
+        }else{
+            redirectToLogin();
+        }
+        return view;
+    }
+    private void redirectToLogin() {
+        Toast.makeText(getContext(), "User need to be logged in!", Toast.LENGTH_SHORT).show();
+        Intent intent = new Intent(getActivity(), LoginActivity.class);
+        startActivity(intent);
+    }
```

Comment:
Several formatting related issues, please resolve

Reply:
made some formatting changes, but couldn't find anything in this highlighted area

Reply:
@iamareebjamal  could you please confirm?